draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,717 @@
+   `aead_nonce`, an empty `aad`, and a `pt` input of `response`. This yields `ct`.
+2. Concatenate `response_nonce` and `ct`, yielding an Encapsulated Response,
+   `enc_response`. Note that `response_nonce` is of fixed length, so there is no
+  ambiguity in parsing either `response_nonce` or `ct`.
+
+In pseudocode, this procedure is as follows:
+
+~~~~~
+secret = context.Export("message/ad-auction-trusted-signals-response", max(Nn, Nk))
+response_nonce = random(max(Nn, Nk))
+salt = concat(enc, response_nonce)
+prk = Extract(salt, secret)
+aead_key = Expand(prk, "key", Nk)
+aead_nonce = Expand(prk, "nonce", Nn)
+ct = Seal(aead_key, aead_nonce, "", response)
+enc_response = concat(response_nonce, ct)
+~~~~~
+
+Clients decrypt an Encapsulated Response by reversing this process. That is,
+Clients first parse `enc_response` into `response_nonce` and `ct`. Then, they
+follow the same process to derive values for `aead_key` and `aead_nonce`, using
+their sending HPKE context, `sctxt`, as the HPKE context, `context`.
+
+The Client uses these values to decrypt `ct` using the AEAD function `Open`.
+Decrypting might produce an error, as follows:
+
+~~~~~
+response, error = Open(aead_key, aead_nonce, "", ct)
+~~~~~
+
+### Framing and Padding {#response-framing}
+
+The plaintext response message uses the framing described in {{framing}}.
+
+Padding is applied with sizes as multiples of 2^n KBs ranging from 0 to 2MB. So the valid response
+sizes will be `[0, 128B, 256B, 512B, 1KB, 2KB, 4KB, 8KB, 16KB, 32KB, 64KB, 128KB, 256KB, 512KB, 1MB,
+2MB]`.
+
+### Response Schema {#response-schema}
+
+The response MAY be compressed. The compression is applied independently to each compression group. That
+means, the response object mainly contains a list of compressed blobs, each for one compression
+group. Each blob is for outputs of one or more partitions, sharing the same `compressionGroup` value
+as specified in the request.
+
+The response is a [CBOR] encoded message with the following [CDDL] schema:
+
+~~~~~ cddl
+response = {
+  ? compressionGroups : [* compressionGroup]
+}
+
+compressionGroup = {
+  ? compressionGroupId: uint,
+  ; Partition outputs with the same `compressionGroupId` specified in the request
+  ; are compressed together.
+  ? ttl_ms: uint,
+  ; Adtech-specified TTL for client-side caching. In milliseconds. Unset means no caching.
+  ? content: bstrs
+  ; Compressed CBOR binary string using the algorithm specified in the request
+  ; For details see compressed response content schema below.
+}
+~~~~~
+
+#### CompressionGroup {#compression-group}
+
+The content of each `compressionGroup` is a serialized [CBOR] list of partition outputs. This object
+contains actual key value results for partitions in the corresponding compression group. The
+uncompressed, deserialized [CBOR] content has the following [CDDL] schema:
+
+~~~~~ cddl
+compressionGroup = [* partitionOutput]
+; Array of PartitionOutput objects
+
+partitionOutput = {
+  ? id: uint
```

Comment:
Should this field be required? I'm not sure what a partitionOutput without an ID would mean.

Reply:
Done.